### PR TITLE
Implement task's list, find, and search command

### LINKED ISSUE #67: As a student, I want to find tasks by date
so that I can keep track of tasks on a particular date

### PR DESCRIPTION
**Pull request by dinhnhobao**
Closes #67

### CODE REVIEW

**Comment by alushingg** [comment]
Is it possible for you to pull first and commit again? Thanks

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 1]
```diff
@@ -34,10 +34,38 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
...
+     * Returns true if the {@code sentence} contains the {@code word}.
+     *   Ignores case, but a full word match is required.
+     *   <br>examples:<pre>
+     *       containsWordIgnoreCase("ABc def", "abc") == true
+     *       containsWordIgnoreCase("ABc def", "DEF") == true
+     *       containsWordIgnoreCase("ABc def", "AB") == false //not a full word match
+     *       </pre>
+     * @param sentence cannot be null
+     * @param word cannot be null, cannot be empty, must be a single word
+     */
```
Comment (zixinn) [inline src/main/java/seedu/address/commons/util/StringUtil.java]: I think you can update the javadoc comment

File: src/main/java/seedu/address/logic/parser/task/TaskSearchCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+
+/**
+ * Parses input arguments and creates a new TaskAddCommand object
+ */
+public class TaskSearchCommandParser implements Parser<TaskSearchCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the TaskAddCommand
+     * and returns a TaskAddCommand object for execution.
```
Comment (zixinn) [inline src/main/java/seedu/address/logic/parser/task/TaskSearchCommandParser.java]: I think can change TaskAddCommand to TaskSearchCommand in the javadoc

**Comment by alushingg** [comment]
Seems ok. Anyway for task add if i input 24:55 for the time. the app will hang because the exception is not catch i guess. it will fail at the constructor in TaskDateTime.

File: src/main/java/seedu/address/commons/util/StringUtil.java (hunk 1 above)
Comment (dinhnhobao) [reply src/main/java/seedu/address/commons/util/StringUtil.java]: Resolved. Thanks!

File: src/main/java/seedu/address/logic/parser/task/TaskSearchCommandParser.java (hunk 2 above)
Comment (dinhnhobao) [reply src/main/java/seedu/address/logic/parser/task/TaskSearchCommandParser.java]: Resolved. Thanks!

**Comment by dinhnhobao** [comment]
> commit

I think I already pulled it.
image

**Comment by dinhnhobao** [comment]
hmm not sure, just now I was resolving some conflicts which involve Lesson. I can check it later.

**Comment by alushingg** [comment]
The ug seems to change to old one

**Pull request merged**